admin/nodes/urls.py

```diff
@@ -3,5 +3,12 @@
 from . import views
 
 urlpatterns = [
-    url(r'^$', views.NodeFormView.as_view(), name='node'),
+    url(r'^$', views.NodeFormView.as_view(),
+        name='search'),
+    url(r'^id-(?P<guid>[a-z0-9]+)/$', views.NodeView.as_view(),
+        name='node'),
+    url(r'^id-(?P<guid>[a-z0-9]+)/remove_node/$', views.remove_node,
```

Comment:
This can go in a follow-up PR, but you should refactor this so that nodes are removed with a `DELETE` request to `/<guid>/` rather than adding this extra `remove_node` route.